server/sonar-server/src/main/java/org/sonar/server/rule/RuleService.java

```diff
@@ -56,20 +55,19 @@ public RuleService(RuleIndex index, RuleUpdater ruleUpdater, RuleCreator ruleCre
   }
 
   @CheckForNull
+  @Deprecated
   public Rule getByKey(RuleKey key) {
-    return index.getNullableByKey(key);
+    throw new UnsupportedOperationException("Please use RuleDao");
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "Please use RuleDao" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1192)